web/src/client/manager.js

```diff
@@ -108,33 +100,37 @@ class ManagerBaseClient {
    * @return {import ("./dbus").RemoveFn} function to disable the callback
    */
   onPhaseChange(handler) {
-    return this.client.onObjectChanged(MANAGER_PATH, MANAGER_IFACE, (changes) => {
-      if ("CurrentInstallationPhase" in changes) {
-        handler(changes.CurrentInstallationPhase.v);
+    return this.client.onEvent("InstallationPhaseChanged", ({ phase }) => {
+      if (phase) {
+        handler(phase);
       }
     });
   }
 
   /**
    * Runs cleanup when installation is done
    */
-  async finishInstallation() {
-    const proxy = await this.client.proxy(MANAGER_IFACE);
-    return proxy.Finish();
+  finishInstallation() {
+    return this.client.post("/manager/install");
   }
 
   /**
    * Returns whether Iguana is used on the system
+   *
+   * @return {Promise<boolean>}
    */
   async useIguana() {
-    const proxy = await this.client.proxy(MANAGER_IFACE);
-    return proxy.IguanaBackend;
+    const installer = await this.client.get("/manager/installer");
+    return installer.iguana;
   }
 }
 
 /**
-  * Client to interact with the Agama manager service
-  */
-class ManagerClient extends WithProgress(WithStatus(ManagerBaseClient, MANAGER_PATH), MANAGER_PATH) { }
+ * Client to interact with the Agama manager service
+ */
+class ManagerClient extends WithProgress(
+  WithStatus(ManagerBaseClient, "/manager/status", "org.opensuse.Agama.Manager1"),
```

Comment:
Why not having a variable for "org.opensuse.Agama.Manager1" too? Is it always in sync with MANAGER_PATH?

I mean, MANAGER_PATH.replace("/", ".") should work always?

Reply:
I don't know if it will work in all cases (apart from the fact that you need to remove the leading `/`).